airbyte-integrations/connectors/source-youtube-analytics/source_youtube_analytics/source.py

```diff
@@ -0,0 +1,184 @@
+        return reports
+
+    def path(
+        self, stream_state: Mapping[str, Any] = None, stream_slice: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> str:
+        if not self.job_id:
+            self.job_id = self.jobs_stream.create(self.name)
+            self.logger.info(f"YouTube reporting job is created: '{self.job_id}'")
+        return "jobs/{}/reports".format(self.job_id)
+
+
+class ChannelReports(HttpSubStream):
+    " https://developers.google.com/youtube/reporting/v1/reports/channel_reports "
+
+    name = None
+    primary_key = None
+    cursor_field = "date"
+    url_base = "https://youtubereporting.googleapis.com/v1/"
+    transformer = TypeTransformer(TransformConfig.DefaultSchemaNormalization)
+
+    def __init__(self, name: str, dimensions: List[str], **kwargs):
+        self.name = name
+        self.primary_key = dimensions
+        return super().__init__(**kwargs)
+
+    def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
+        return None
+
+    def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
+        fp = io.StringIO(response.text)
+        reader = csv.DictReader(fp)
+        for record in reader:
+            yield record
+
+    def get_updated_state(self, current_stream_state: MutableMapping[str, Any], latest_record: Mapping[str, Any]) -> Mapping[str, Any]:
+        if not current_stream_state:
+            return {self.cursor_field: latest_record[self.cursor_field]}
+        return {self.cursor_field: max(current_stream_state[self.cursor_field], latest_record[self.cursor_field])}
+
+    def path(
+        self, stream_state: Mapping[str, Any] = None, stream_slice: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> str:
+        return stream_slice["parent"]["downloadUrl"][len(self.url_base):]
+
+    def read_records(self, *, stream_slice: Mapping[str, Any] = None, **kwargs) -> Iterable[Mapping[str, Any]]:
+        parent = stream_slice.get("parent")
+        if parent:
+            yield from super().read_records(stream_slice=stream_slice, **kwargs)
+        else:
+            yield from []
+
+
+class SourceYoutubeAnalytics(AbstractSource):
+    @staticmethod
+    def get_authenticator(config):
+        credentials = config["credentials"]
+        client_id = credentials["client_id"]
+        client_secret = credentials["client_secret"]
+        refresh_token = credentials["refresh_token"]
+
+        return Oauth2Authenticator(
+            token_refresh_endpoint="https://oauth2.googleapis.com/token",
+            client_id=client_id,
+            client_secret=client_secret,
+            refresh_token=refresh_token,
+        )
+
+    def check_connection(self, logger, config) -> Tuple[bool, any]:
+        authenticator = self.get_authenticator(config)
+        jobs_stream = JobsStream(authenticator=authenticator)
+
+        try:
+            jobs_stream.list()
```

Comment:
Are we receiving some sort of records here? If there are more than 1 records in the output, we can use `next()` to process only 1 record message, this should be enough for the check method.

Reply:
it make one HTTP request return one json response `{'jobs': [job1, job2, job3, jobN]}`
I cannot wrap `next` because it's not iterator 

https://github.com/airbytehq/airbyte/blob/c3330d257def9cd4fe5f199736213664b333240a/airbyte-integrations/connectors/source-youtube-analytics/source_youtube_analytics/source.py#L43